boa/src/builtins/math/mod.rs

```diff
@@ -478,7 +475,7 @@ pub fn sin(_: &Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
 ///
 /// [spec]: https://tc39.es/ecma262/#sec-math.sinh
 /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/sinh
-pub fn sinh(_: &Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
+pub fn sinh(_: &mut Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
```

Comment:
Just for my understanding, why do we need all these to be mutable?

Reply:
The first argument `this` can be altered by Constructor functions. 
Its up to them to change the `this` value however they want.

Usually `this` starts out being a plain ordinary JS object, and String() will add a [[StringData]] internal slot to it. This mutates the value.

Why this wasn't a compile error before i don't know, but something in my refactoring meant that rust now moans if you mutate it.
imo it should have always been an issue but we got away with it before.

Reply:
For regular functions it doesn't need to be mutable, as the first argument is a reference to the function itself (wrapped in a Value).
I think in future we can have 2 separate signatures for Construct and Func, for now they use the same.